### PR TITLE
[#IOPID-1449] enable `strict=true` on typescript config

### PR DESCRIPTION
**Pull request by BurnedMarshal**
#### List of Changes

Code refactory after enabling the `strict=true` option in `tsconfig`.

#### Motivation and Context

Having a `strict=false` option for Typescript transpiling can generate application and logic errors to handle properly `undefined` or `null` properties. This setting helps to prevent bugs in production.

#### How Has This Been Tested?

`build ` and unit test

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: GetServicePreferences/handler.ts [hunk 1]
```diff
@@ -125,6 +146,8 @@ const getUserServicePreferencesOrDefault = (
                 return toDefaultEnabledUserServicePreference(version);
               case ServicesPreferencesModeEnum.MANUAL:
                 return toDefaultDisabledUserServicePreference(version);
+              default:
+                return void 0 as never;
```
Comment (arcogabbo) [inline GetServicePreferences/handler.ts:150]: is this needed? project compiles without the default case because all possible values are already addressed

**Review by arcogabbo — APPROVED** [review]

**Review by gquadrati — COMMENTED** [review]
Small comments - Part 1 :)

File: GetServicePreferences/handler.ts (hunk 1 above)
Comment (gquadrati) [reply GetServicePreferences/handler.ts:150]: Adding this default case could be beneficial, particularly if enhanced with [Exhaustiveness Checking](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking)

File: UpsertedProfileOrchestrator/handler.ts [hunk 2]
```diff
@@ -105,14 +105,27 @@ export const getUpsertedProfileOrchestratorHandler = (params: {
...
 
     // Check if the profile email is changed
     const isProfileEmailChanged =
-      profileOperation === "UPDATED" && newProfile.email !== oldProfile.email;
+      profileOperation.type === "UPDATED" &&
+      newProfile.email !== profileOperation.oldProfile.email;
     // NOTE: if the following check is changed make sure to pass add the name field
     // to CreateProfile method which also calls this orchestrator
-    if (isProfileEmailChanged) {
+    if (isProfileEmailChanged && newProfile.email) {
```
Comment (gquadrati) [inline UpsertedProfileOrchestrator/handler.ts:128]: Although it's unlikely, if the email changes to undefined, could we raise an event?

**Review by gquadrati — COMMENTED** [review]
Small comments - Part 2/2.

File: utils/config.ts [hunk 3]
```diff
@@ -71,7 +71,25 @@ export const VisibleServiceConfig = t.interface({
...
+        ? t.success(JSON.parse(value))
+        : t.failure(value, context);
+    } catch (err) {
+      return t.failure(value, context);
+    }
+  },
+  JsonFromString.encode
+);
+
+export const BetaUsersFromString = withFallback(SafeJsonFromString, []).pipe(
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply utils/config.ts:92]: Thanks, your suggestion works fine

File: UpsertedProfileOrchestrator/handler.ts (hunk 2 above)
Comment (BurnedMarshal) [reply UpsertedProfileOrchestrator/handler.ts:128]: Do you mean just for monitoring?

File: utils/config.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply utils/config.ts]: Fixed in 76cc677

File: UpdateProfile/handler.ts [hunk 4]
```diff
@@ -215,7 +215,7 @@ export function UpdateProfileHandler(
     const profile = apiProfileToProfile(
       profilePayload,
       fiscalCode,
-      emailChanged ? false : existingProfile.isEmailValidated,
+      emailChanged ? false : existingProfile.isEmailValidated || false,
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply UpdateProfile/handler.ts]: Fixed in 9967527. Thanks is definitively more accurate.

File: utils/storage.ts [hunk 5]
```diff
@@ -32,7 +32,7 @@ export const insertTableEntity = (
         resolve(
           // We need to check error first because response could be null
           // @ref https://github.com/Azure/azure-storage-node/blob/v2.10.2/lib/common/services/storageserviceclient.js#L250
-          error || !response.isSuccessful
+          error || !response?.isSuccessful
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply utils/storage.ts:35]: the test already exists, is the error null and result null scenario in here https://github.com/pagopa/io-functions-app/blob/IOPID-1449-strict-true/utils/__tests__/storage.test.ts#L35

File: utils/storage.ts [hunk 6]
```diff
@@ -61,7 +61,7 @@ export const deleteTableEntity = (
...
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply utils/storage.ts:64]: ditto
https://github.com/pagopa/io-functions-app/blob/IOPID-1449-strict-true/utils/__tests__/storage.test.ts#L58

[the remark this answers is not in the document]
Comment (gquadrati) [reply utils/storage.ts:64]: I initially considered a test where both values are undefined, but upon reflection, I realized its lack of value.

**Review by gquadrati — APPROVED** [review]
🚀

**Pull request merged**